cpp/src/arrow/util/decimal_test.cc

```diff
@@ -329,7 +313,8 @@ TEST_P(Decimal128ParsingTest, Parse) {
   std::tie(test_string, expected_low_bits, expected_scale) = GetParam();
   Decimal128 value;
   int32_t scale;
-  ASSERT_OK(Decimal128::FromString(test_string, &value, NULLPTR, &scale));
+  ASSERT_OK_AND_ASSIGN(value,
+                       Decimal128::FromString(test_string, (int32_t*)nullptr, &scale));
```

Comment:
I need to cast `nullptr` to `int32_t*` explicitly because both Result-returning and Status-returning versions of `Decimal128::FromString` are ambiguous when they called with three arguments.
I guess this introduces source-level incompatibility.

@pitrou Do you think how should we handle this problem?